packages/contracts-communication/contracts/modules/InterchainModule.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {InterchainModuleEvents} from "../events/InterchainModuleEvents.sol";
+import {IInterchainDB} from "../interfaces/IInterchainDB.sol";
+import {IInterchainModule} from "../interfaces/IInterchainModule.sol";
+
+import {InterchainEntry} from "../libs/InterchainEntry.sol";
+
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+
+/// @notice Common logic for all Interchain Modules.
+abstract contract InterchainModule is InterchainModuleEvents, IInterchainModule {
+    address public immutable INTERCHAIN_DB;
+
+    constructor(address interchainDB) {
+        INTERCHAIN_DB = interchainDB;
+    }
+
+    /// @inheritdoc IInterchainModule
+    function requestVerification(uint256 destChainId, InterchainEntry memory entry) external payable {
+        if (msg.sender != INTERCHAIN_DB) {
```

Comment:
Noting that if interchainDB address ever changes, all modules will need to re-deploy

Reply:
That's the intended behavior imo, as the module should be compatible only with the latest InterchainDB contract.

In the unlikely event that InterchainDB needs to be updated and redeployed, its design shouldn't be constrained by the existing DB <> Module interaction workflow.